source/intel.js

```diff
@@ -100,10 +100,50 @@ const display_tech_trading = () => {
 	}
 	return tech_trade_screen
 }
+let cached_events = [];
+
+const update_event_cache = (fetchSize) => {
```

Comment:
`update_event_cache` has the inherit callback `recieve_new_messages` which we should detail in a comment.

Reply:
Or we place the callback in the function definition:
```
const update_event_cache = (fetchSize,callback) => {
```
Then update lines `141` and `217` for calling `update_event_cache`. If we can have a default callback set it to `recieve_new_messages`.
```
const update_event_cache = (fetchSize,callback=recieve_new_messages) => {
```

Reply:
I'd be happy to pass both the successCallback and the errorCallback. It will also make the short-lived test against the full load code simple to write.

Reply:
I have made this update, as well as written the cache validation code.

In my opinion this PR is ready to land. As a separate issue, I think someone should fix the bug about having the 100,000 hardcoded, but as it's been working fine this way for ages to me it seems simplest to commit the caching code as is, see that that works, and add the complexity for the big constant in a later change.